ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java

```diff
@@ -306,6 +309,10 @@ private void investigateDialect() {
             database = identifyDialect(database);
             EjbLogger.EJB3_TIMER_LOGGER.debugf("New Database dialect is '%s'.", database);
         }
+
+        if (database == null) {
+            throw EjbLogger.EJB3_TIMER_LOGGER.failedToFindDialect();
+        }
     }
```

Comment:
With a null database value, the timer persistence should still be able to function. So it should not throw exception here. Logging a warning like in line 303 should suffice. We can move line 302-304 to here to handle this case for all cases.

Reply:
Around line 245, we can add a null check to prevent NPE from `database` field:

```java
  if(database != null) {
        switch (database) {
            case DB2:
                adjustCreateAutoTimerStatement("FROM SYSIBM.SysDummy1 ");
            break;
            case ORACLE:
                adjustCreateAutoTimerStatement("FROM DUAL ");
            break;
        }
  }
```

Reply:
I see, I figured it wouldn't work correctly if e.g. the `adjustCreateAutoTimerStatement` wasn't called. I've refactored the PR.